src/orion/core/worker/primary_algo.py

```diff
@@ -79,64 +119,133 @@ def suggest(self, num):
         if transformed_trials is None:
             return None
 
-        for trial in transformed_trials:
-            self._verify_trial(trial, space=self.transformed_space)
-            if trial not in self.transformed_space:
+        trials: list[Trial] = []
+        for transformed_trial in transformed_trials:
+            if transformed_trial not in self.transformed_space:
                 raise ValueError(
-                    f"Trial {trial.id} not contained in space:"
-                    f"\nParams: {trial.params}\n: Space{self.transformed_space}"
+                    f"Trial {transformed_trial.id} not contained in space:\n"
+                    f"Params: {transformed_trial.params}\n"
+                    f"Space: {self.transformed_space}"
                 )
-
-        trials = []
-        for transformed_trial in transformed_trials:
-            trial = self.transformed_space.reverse(transformed_trial)
-            self._verify_trial(trial, space=self.space)
-            trials.append(trial)
-
+            original = self.transformed_space.reverse(transformed_trial)
+            if original in self.registry:
+                # We already have a trial that is equivalent to this one.
+                # Copy over the status and results and observe it.
+                transformed_trial = _copy_status_and_results(
+                    original_trial=original, transformed_trial=transformed_trial
```

Comment:
Shouldn't we fetch the original from registry to do this? `original` is a transformed trial that was suggested from the algo, so it should not have results. `self.register[_get_id(original.id)]` could have results and a different status however.

Reply:
Fixed in eebe53ca24d35db5b9cb917c3a319e4755d41266

Reply:
We should make sure to add a test for this since the issue was not detected.

Reply:
Agreed. Do you already have an idea of what this test would do, by any chance?

Here's what I'm thinking so far:
- Setup the algo with some completed trials.
- Get a suggested trial from the algo, and somehow make it match an existing trial when translated back into the original space (e.g. through a pretty aggressive `Precision` transform?)
- Make sure that the algo observes the trial it suggested, with the results and status of the existing trial.

Reply:
Sounds like a plan! :smile:

Reply:
Can we mark this as resolved?